src/flag_gems/ops/stack.py

```diff
@@ -0,0 +1,57 @@
+import itertools
+import logging
+from typing import List, Tuple, Union
+
+import torch
+import triton
+
+from ..utils import pointwise_dynamic
+from ..utils.tensor_wrapper import StridedBuffer
+
+
+@pointwise_dynamic(is_tensor=[True], promotion_methods=[(0, "DEFAULT")])
+@triton.jit
+def copy_func(x):
+    return x
+
+
+def stack(
+    tensors: Union[Tuple[torch.Tensor, ...], List[torch.Tensor]], dim: int = 0
+) -> torch.Tensor:
+    logging.debug("GEMS STACK")
+
+    if len(tensors) == 0:
+        raise RuntimeError("stack expected a non-empty TensorList")
+
+    inp_shapes = [list(_.shape) for _ in tensors]
+    inp0_shape = inp_shapes[0]
+    for i, s in enumerate(inp_shapes[1:]):
```

Comment:
it does be necessary to do those checks，but it will result in bad performance，I wonder if we can reuse OP checks in `aten` lib

Reply:
Can you give an example?

Reply:
It hit me just when I review this PR, maybe we can think it out togeter in the following PR ：）